modules/javafx.graphics/src/main/java/com/sun/javafx/sg/prism/NGShape3D.java

```diff
@@ -173,7 +181,10 @@ private void renderMeshView(Graphics g) {
         // TODO: 3D Required for D3D implementation of lights, which is limited to 3
         while (pointLightIdx < 3) {
                 // Reset any previously set lights
-                meshView.setPointLight(pointLightIdx++, 0, 0, 0, 0, 0, 0, 0);
+                meshView.setPointLight(pointLightIdx++,
+                        0, 0, 0, // x y z
+                        0, 0, 0, 0, // r g b w
+                        1, 0, 0, 0); // ca la qa maxRange
```

Comment:
Minor: maybe use the getDefaultXxxx methods of NGPointLight?

Reply:
The only impact this has is that the range will be maximal instead of 0. When these reach the shader, they will run the lighting computation as opposed to skipping it. I'm not sure if this will have any performance impact though.

Reply:
In that case, it seems like a generally useful optimization (not just at initialization) to send down `maxRange` as 0 whenever `ca`, `la`, and `qa` are all at their default values.

Reply:
These lights are fillers for the native lights array in the case where the user set less than 3 lights, they are not supposed to do anything, and their color is black. A range of 0 will guarantee that the computation in the shader will be skipped.

In a WIP that I have locally I have removed the "3 lights constraint" altogether, so this part will change in that proposed patch anyway.